src/_pytest/config/__init__.py

```diff
@@ -1662,6 +1663,54 @@ def _warn_about_skipped_plugins(self) -> None:
             )
 
 
+class OutputVerbosity:
+    DEFAULT = "auto"
+    _option_name_fmt = "verbosity_{}"
+
+    def __init__(self, config: Config) -> None:
+        self._config = config
+
+    @property
+    def verbose(self) -> int:
+        """Application wide verbosity level."""
+        return cast(int, self._config.option.verbose)
+
+    def verbosity_for(self, output_type: str) -> int:
```

Comment:
I think if one passes an invalid `output_type` at the moment, we will get an error (dur to `self._config.getini`).

We probably want to have this classs accessible to plugins and users, via `config.output_verbosity`, so we need to think about its API carefully as we will need to support this officially moving on.

Some points:

1. In tandem with pytest's direction regarding type safety, I think we should use an `Enum` with the output types currently available, something along the lines of:

   ```python
   class VerbosityType(Enum):
       """..."""     

       Global = "global"
       Assertions = "assertions"
   ```

   Instead of a simple `output_type: str` parameter/value.

2. The class name is `OutputVerbosity` and the public methods are `verbosity` and `verbosity_for`, which read a bit redundant: `confg.output_verbosity.verbosity`, `confg.output_verbosity.get_verbosity_for`. I suggest we use simply `get`, defaulting to the `Global` verbosity:

   ```python
   def get(self, type: VerbosityType = VerbosityType.Global) -> int:
       ...
   ```

   Which reads well I think:

   ```python
   config.output_verbosity.get()
   config.output_verbosity.get(VerbosityType.Assertions)
   ```

3. We need to properly document both `OutputVerbosity` and `VerbosityType` in the reference docs (including some usage examples, but possibly we can write those in the class docstring). We also need to expose `VerbosityType` in the `pytest` namespace.

4. Given `OutputVerbosity` is public, we should make `OutputVerbosity.addini` private, as it is intended for internal use only.


I'm not married to the names above, so suggestions and bikeshedding are welcome.

Reply:
1. I opted for `str` because I was thinking that plugins might want also register types, like any other config setting. Which would mean that it wouldn't be possible to statically define all of the types. However, reflecting on that, I'm not sure that level of extensibility is necessary. It is likely better for `pytest` to say "here are N buckets that a plugin can utilize" and that way users of multiple plugins that leverage this functionality will see consistent behavior without needing to enable M settings.
2. That is fair. It also works nicely with the enum suggestion.
3. Sure
4. I had it as public for the reasons described in 1, but it makes sense to make it private based on that change.